README.md

```diff
@@ -363,6 +366,213 @@ async function main() {
 main().catch(console.error);
 ```
 
+## Workload Identity Federation
```

Comment:
This is a _massive_ addition to the README, and I worry a little about the size.  I am ok keeping it here for now (non-blocking feedback), but we need a better documentation strategy for auth.  @tbpg

Reply:
@bcoe also pointed out the same issue which I agree with. There are larger plans to update the official authentication docs: https://cloud.google.com/docs/authentication. Though this could be delayed until Q3. Currently, they are only updating the workload identity federation docs:

- https://cloud.google.com/iam/docs/access-resources-aws
- https://cloud.google.com/iam/docs/access-resources-azure
- https://cloud.google.com/iam/docs/access-resources-oidc

It would be a good idea to align with the overall documentation effort in the cloud docs and the GitHub repos.